commcare_connect/opportunity/visit_import.py

```diff
@@ -114,12 +119,17 @@ def _bulk_update_visit_status(opportunity: Opportunity, dataset: Dataset):
 
             UserVisit.objects.bulk_update(to_update, fields=["status", "reason"])
             missing_visits |= set(visit_batch) - seen_visits
+    if opportunity.auto_approve_payments:
+        approve_completed_work_and_update_payment_accrued.delay(list(seen_completed_works))
```

Comment:
Is the accrual logic the same here? If so you can call the function from in the task. Or just call it outside the if/else block since it happens in either case.

Relatedly, why does this need to be a separate task? The upload is already happening in a celery task.

Reply:
> Is the accrual logic the same here? If so you can call the function from in the task. Or just call it outside the if/else block since it happens in either case.
The payment accrual logic is same, the task only adds the logic to approve a completed work if it has no flags. I will simplify this code to run the update payment accrued function only once and do the completed work approval in the `if` block.

> Relatedly, why does this need to be a separate task? The upload is already happening in a celery task.
I checked the import does not happen in a task only the export for files is done through a celery task.